drivers/vulkan/vulkan_context.cpp

```diff
@@ -1680,7 +1680,10 @@ Error VulkanContext::_create_semaphores() {
 		/*pNext*/ nullptr,
 		/*flags*/ VK_FENCE_CREATE_SIGNALED_BIT
 	};
-	for (uint32_t i = 0; i < FRAME_LAG; i++) {
+
+	CRASH_COND_MSG(frame_count == 0, "Used before initialization.");
```

Comment:
Wonderful! I ran into that same issue trying a provisional fix to what your PR does, this seems much cleaner than the current logic on master.